### PR TITLE
feat: Implement loading indicator component

### PR DESCRIPTION
**Pull request by TSenter**

### CODE REVIEW

File: packages/ember/src/components/loading-indicator.gts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export default class LoadingIndicatorComponent extends Component<LoadingIndicatorSignature> {
+  get typeClass() {
+    const { type } = this.args;
+    assert(
+      'type must be either `border` or `growing`',
+      type === undefined || type === typeBorder || type === typeGrowing,
+    );
+
+    if (type === typeGrowing) {
+      return 'spinner-growing';
```
Comment (lnicely) [inline packages/ember/src/components/loading-indicator.gts:25]:
```suggestion
      return 'spinner-grow';
```

**Review by lnicely — CHANGES_REQUESTED** [review]

File: packages/ember/src/components/loading-indicator.gts (hunk 1 above)
Comment (TSenter) [reply packages/ember/src/components/loading-indicator.gts]: Don't know how I missed that! I blame Copilot 😄

**Review by KeithClinard — APPROVED** [review]

File: packages/ember/src/components/loading-indicator.gts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+
+export const typeBorder = 'border';
+export const typeGrowing = 'growing';
```
Comment (lnicely) [inline packages/ember/src/components/loading-indicator.gts:5]:
```suggestion
export const typeGrow = 'grow';
```

File: packages/ember/src/components/loading-indicator.gts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    type?: typeof typeBorder | typeof typeGrowing;
+  };
+}
+
+export default class LoadingIndicatorComponent extends Component<LoadingIndicatorSignature> {
+  get typeClass() {
+    const { type } = this.args;
+    assert(
+      'type must be either `border` or `growing`',
+      type === undefined || type === typeBorder || type === typeGrowing,
```
Comment (lnicely) [inline packages/ember/src/components/loading-indicator.gts:21]:
```suggestion
      'type must be either `border` or `grow`',
      type === undefined || type === typeBorder || type === typeGrow,
```

File: packages/ember/src/components/loading-indicator.gts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    );
+
+    if (type === typeGrowing) {
```
Comment (lnicely) [inline packages/ember/src/components/loading-indicator.gts:24]:
```suggestion
    if (type === typeGrow) {
```

**Review by lnicely — CHANGES_REQUESTED** [review]

**Review by lnicely — APPROVED** [review]

**Review by KeithClinard — APPROVED** [review]

**Pull request merged**